src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs

```diff
@@ -596,8 +596,11 @@ public void Delete(string id, string expectedChangeVector)
                         DocumentsByEntity.Remove(documentInfo.Entity);
                     }
 
-                    DocumentsById.Remove(id);
-                    changeVector = documentInfo.ChangeVector;
+                    using (documentInfo)
```

Comment:
A user may return a `BlittableDocument` and keep it around after calling `Delete()`.
Not likely to be really a scenario to be worried about, but should add that to the breaking changes list?

Reply:
> How likely that is to be actually something meaningful?

Problem with long-lived sessions. Please check the YT issue.

Reply:
Other option is to only do that on Evict and nowhere else? Thoughts?

Reply:
Just add a breaking change note and that should be it, I think.